### PR TITLE
fahrplan: init at 1.1.2

### PR DESCRIPTION
**Pull request by p-h**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/fa/fahrplan/package.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{ lib, python3, fetchFromGitHub }:
+let
+  pname = "fahrplan";
+  owner = "dbrgn";
+  version = "1.1.2";
+in python3.pkgs.buildPythonApplication {
+  inherit pname version;
+  format = "setuptools";
```
Comment (dotlambda) [inline pkgs/by-name/fa/fahrplan/package.nix:8]:
```suggestion
  pyproject = true;
```
and specify `build-system`

File: pkgs/by-name/fa/fahrplan/package.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  src = fetchFromGitHub {
+    inherit owner;
+    repo = pname;
+    rev = "refs/tags/v${version}";
+    hash = "sha256-2QVaA2zqqFuxjkKbDXo+SNzkrDCxXiSYMUOftQ++wO4=";
+  };
+
+  propagatedBuildInputs = with python3.pkgs; [ dateutil requests texttable ];
```
Comment (dotlambda) [inline pkgs/by-name/fa/fahrplan/package.nix:17]:
```suggestion
  dependencies = with python3.pkgs; [ dateutil requests texttable ];
```

File: pkgs/by-name/fa/fahrplan/package.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  pname = "fahrplan";
+  owner = "dbrgn";
+  version = "1.1.2";
+in python3.pkgs.buildPythonApplication {
+  inherit pname version;
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    inherit owner;
+    repo = pname;
```
Comment (dotlambda) [inline pkgs/by-name/fa/fahrplan/package.nix:12]: #277994

File: pkgs/by-name/fa/fahrplan/package.nix [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+let
+  pname = "fahrplan";
+  owner = "dbrgn";
+  version = "1.1.2";
+in python3.pkgs.buildPythonApplication {
+  inherit pname version;
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    inherit owner;
```
Comment (dotlambda) [inline pkgs/by-name/fa/fahrplan/package.nix:11]:
```suggestion
    owner = "dbrgn";
```

File: pkgs/by-name/fa/fahrplan/package.nix [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    repo = pname;
+    rev = "refs/tags/v${version}";
+    hash = "sha256-2QVaA2zqqFuxjkKbDXo+SNzkrDCxXiSYMUOftQ++wO4=";
+  };
+
+  propagatedBuildInputs = with python3.pkgs; [ dateutil requests texttable ];
+
+  meta = with lib; rec {
```
Comment (dotlambda) [inline pkgs/by-name/fa/fahrplan/package.nix:19]:
```suggestion
  meta = {
```
https://github.com/NixOS/nixpkgs/issues/208242

File: pkgs/by-name/fa/fahrplan/package.nix [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    description =
+      "Command line access to the SBB/CFF/FFS timetable from with human readable argument parsing";
```
Comment (dotlambda) [inline pkgs/by-name/fa/fahrplan/package.nix:21]:
```suggestion
      "Command line access to the SBB/CFF/FFS timetable with human readable argument parsing";
```
or anything else that's grammatically correct

**Review by dotlambda — COMMENTED** [review]
nixfmt

File: pkgs/by-name/fa/fahrplan/package.nix (hunk 2 above)
Comment (p-h) [reply pkgs/by-name/fa/fahrplan/package.nix]: Out of curiosity, why is it done this way? The docs say to use propagatedBuildInputs

Comment (dotlambda) [reply pkgs/by-name/fa/fahrplan/package.nix]: No they don't: https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#buildpythonpackage-parameters-buildpythonpackage-parameters

File: pkgs/by-name/fa/fahrplan/package.nix [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    setuptools
+  ];
+
+  dependencies = with python3.pkgs; [
+    dateutil
+    requests
+    texttable
+  ];
+
+  meta = with lib; {
```
Comment (dotlambda) [inline pkgs/by-name/fa/fahrplan/package.nix:31]:
```suggestion
  meta = {
```

File: pkgs/by-name/fa/fahrplan/package.nix [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    description = "Command line access to the SBB/CFF/FFS timetable with human readable argument parsing";
+    homepage = "https://github.com/dbrgn/fahrplan";
+    changelog = "https://github.com/dbrgn/fahrplan/blob/${version}/CHANGELOG.md";
```
Comment (dotlambda) [inline pkgs/by-name/fa/fahrplan/package.nix:34]:
```suggestion
    changelog = "https://github.com/dbrgn/fahrplan/blob/${src.rev}/CHANGELOG.md";
```

File: pkgs/by-name/fa/fahrplan/package.nix [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+    rev = "refs/tags/v${version}";
+    hash = "sha256-2QVaA2zqqFuxjkKbDXo+SNzkrDCxXiSYMUOftQ++wO4=";
+  };
+
+  build-system = with python3.pkgs; [
+    setuptools
+  ];
+
+  dependencies = with python3.pkgs; [
+    dateutil
```
Comment (dotlambda) [inline pkgs/by-name/fa/fahrplan/package.nix:23]:
```suggestion
    python-dateutil
```
`dateutil` is an alias

**Review by FliegendeWurst — APPROVED** [review]

**Pull request merged**